src/Dapr.AspNetCore/DaprEndpointRouteBuilderExtensions.cs

```diff
@@ -84,12 +86,52 @@ private static IEndpointConventionBuilder CreateSubscribeEndPoint(IEndpointRoute
                         .Select(part => part.Content))));
 
                     var rawPayload = entry.EnableRawPayload ?? options?.EnableRawPayload;
-                    var subscription = new Subscription
+                    var subscriptionKey = entry.PubsubName + ":" + entry.Name;
```

Comment:
You can use a tuple for this, and it's a little more clear that it's a compound key. 

```
var subscriptionKey = (entry.PubsubName, entry.Name);
```

Make the dictionary a `Dictionary<(string, string), Subscription>`